src/frontend/src/store/index.ts

```diff
@@ -108,8 +112,64 @@ export default new Vuex.Store({
     MUTATATION_SET_MINING(state, mining) {
       state.mining = mining;
     },
+    MUTATATION_SET_EDITING(state, editing) {
+      state.editing = editing;
+    },
+    MUTATATION_SET_FIRST_NAME(state, firstName) {
+      state.firstName = firstName;
+    },
+    MUTATATION_SET_LAST_NAME(state, lastName) {
+      state.lastName = lastName;
+    },
+    MUTATATION_SET_PASSWORD(state, password) {
+      state.password = password;
+    },
   },
   actions: {
+    ACTION_UPDATE_USER({ commit, getters }, values) {
+      commit("MUTATATION_SET_LOADING", true);
+
+      const password = values.password ? values.password : getters.password;
+      const firstName = values.firstName ? values.firstName : getters.firstName;
+      const lastName = values.lastName ? values.lastName : getters.lastName;
+
+      axios
+        .post("http://localhost/users/update", {
+          "first_name": firstName,
+          "last_name": lastName,
+          "umnetID": getters.umnetId,
+          "public_key": getters.walletId,
+          "curr_password": getters.password,
+          "new_password": password
+        })
+        .then(
+          () => {
+            commit("MUTATION_SET_WALLET_CREATED", true);
+            commit("MUTATATION_SET_LOADING", false);
+            commit("MUTATATION_SET_FIRST_NAME", firstName);
+            commit("MUTATATION_SET_LAST_NAME", lastName);
+            commit("MUTATATION_SET_PASSWORD", password);
+
+
+          },
+          (err) => {
+            commit("MUTATATION_SET_LOADING", false);
+            Vue.$toast.error(
+              err.response.data.error
+                ? err.response.data.error
+                : "An error occurred. Please try again.",
```

Comment:
I think we can create a constant for this string since we use it a lot for errors!